### PR TITLE
fix(circuit-relay): respect applyDefaultLimit when it is false

### PR DESCRIPTION
**Pull request by achingbrain**
## Description

We were accidentally applying the default limits if no limits were set, which is intentionally the case when the `applyDefaultLimit` config option is set to `false`.

Also simplifies the relay test setup to create nodes with just config overrides to remove a bit of the copy/paste.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation if necessary (this includes comments as well)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/libp2p/src/circuit-relay/utils.ts [hunk 1]
```diff
@@ -13,6 +12,8 @@ import type { Uint8ArrayList } from 'uint8arraylist'
 const log = logger('libp2p:circuit-relay:utils')
 
 async function * countStreamBytes (source: Source<Uint8Array | Uint8ArrayList>, limit: { remaining: bigint }): AsyncGenerator<Uint8Array | Uint8ArrayList, void, unknown> {
```
Comment (maschad) [inline packages/libp2p/src/circuit-relay/utils.ts:14]: we probably should change the signature of this function to

```suggestion
async function * countStreamBytes (source: Source<Uint8Array | Uint8ArrayList>, limitBytes: bigint): AsyncGenerator<Uint8Array | Uint8ArrayList, void, unknown> {
```

and replace all the `limit.remaining`

File: packages/libp2p/src/circuit-relay/utils.ts [hunk 2]
```diff
@@ -47,25 +48,33 @@ const doRelay = (src: Stream, dst: Stream, abortSignal: AbortSignal, limit: Requ
...
   let srcDstFinished = false
   let dstSrcFinished = false
 
-  const dataLimit = {
-    remaining: limit.data
+  let dataLimit: { remaining: bigint } | undefined
+
+  if (limit?.data != null) {
+    dataLimit = {
+      remaining: limit.data
```
Comment (maschad) [inline packages/libp2p/src/circuit-relay/utils.ts:66]: See comment above, If the function signature is changed then creating this object is unnecessary

**Review by maschad — CHANGES_REQUESTED** [review]
Good catch, I think we can do some refactors and then land it.

File: packages/libp2p/src/circuit-relay/utils.ts (hunk 1 above)
Comment (achingbrain) [reply packages/libp2p/src/circuit-relay/utils.ts:14]: Unfortunately this doesn't work and would introduce a new bug.

The data limit applies in both directions (e.g. `A->Relay->B` and `B->Relay->A`) so both ends of the stream need to decrement the same `remaining` variable.

Passing an object copies by reference, passing a number would copy by value so you'd end up with separate limits for each direction.

**Review by maschad — APPROVED** [review]

**Pull request merged**